### ISSUE TITLE
Integrate existing Kotlin support to Buildship and let people try it out

### ISSUE THREAD

**Issue by donat** [eclipse/buildship#222]
<a href="https://github.com/donat"><img src="https://avatars.githubusercontent.com/u/419883?v=3" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [donat](https://github.com/donat)**
_Friday Jan 13, 2017 at 10:35 GMT_

----

## Code review

- [ ] Does the change work?
- [ ] Is the change sufficiently tested?
- [ ] Is the public API modification handled appropriately?
- [ ] Is it sufficiently documented?
- [ ] Is it implemented well?

See also the [full checklist](https://github.com/gradle/process/wiki/Process#implementation-checklist).

**Comment by donat** [comment]
<a href="https://github.com/donat"><img src="https://avatars.githubusercontent.com/u/419883?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [donat](https://github.com/donat)**
_Friday Jan 27, 2017 at 14:16 GMT_

----

Here's how to try it:
1. Download [Eclipse Neon](https://www.eclipse.org/downloads/packages/eclipse-ide-java-ee-developers/neon2)
2. Install the Kotlin Eclipse plugins. From the Eclipse menu select `Help > Install New Software` and use this update site: http://builds.gradle.org:8000/eclipse/update-site/mirror/kotlin-eclipse-dev/
3. Install the Buildship snapshot from this update site: https://builds.gradle.org/repository/download/Tooling_Master_Eclipse_Experimental_Eclipse46withKotlinSupport/2195628:id/update-site The credentials are `guest/guest`
4. Import the samples from the `gradle-script-kotlin` library.

What to expect:
- Syntax highlighting, error markers, navigation, refactoring, code completion works.
- Kotlin model accessed synchronously, therefore the UI hang a lot.
- Javadoc hover doesn't work.

**Comment by donat** [comment]
<a href="https://github.com/donat"><img src="https://avatars.githubusercontent.com/u/419883?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [donat](https://github.com/donat)**
_Friday Jan 27, 2017 at 14:21 GMT_

----

PR for the review: https://github.com/eclipse/buildship/pull/202

**Comment by donat** [comment]
<a href="https://github.com/oehme"><img src="https://avatars.githubusercontent.com/u/1191797?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [oehme](https://github.com/oehme)**
_Monday Jan 30, 2017 at 17:36 GMT_

----

I gave it a first spin. I could open a Kotlin script and write some very basic stuff, yay :) However, there seem to be a lot of problems atm (or maybe my install is broken somehow):

- it doesn't use the specified Gradle version, but a hardcoded one
- loading the model seems pretty slow for such small projects
- loading the model every time I open a file is making the slowness worse
- I get lots of error markers in multi-kotlin-project-config-injection example
  - Unresolved reference: KotlinCompile
  - Expression 'plugins' of type 'PluginContainer!' cannot be invoked as a function. The function 'invoke()' is not found
  - cannot access class java.lang.Object, check your module path
  - and many more, I can do a screenshare if needed
- auto completion is generally very slow
- navigation to source doesn't work
- no outline view

I didn't get further than that, since the general slowness and the many error markers make it hard to write something.

**Comment by donat** [comment]
<a href="https://github.com/donat"><img src="https://avatars.githubusercontent.com/u/419883?v=3" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [donat](https://github.com/donat)**
_Tuesday Jan 31, 2017 at 09:53 GMT_

----

Thanks for sparing time for the evaluation. No, it's not broken, it's just a prototype :)

**Comment by calvertdw** [comment]
![image](https://cloud.githubusercontent.com/assets/1161586/26012082/ef00d840-3718-11e7-9276-7a9f5db11dfe.png)

Java model doesn't load for me.

**Comment by donat** [comment]
The Kotlin build script support is now functional (e.g. loads the correct classpath). It's time to give it another spin and see what's missing/not working.

**Comment by Golvanig** [comment]
Hello, 
Thanks for all this effort. If I can help by doing some tests...
We'll be happy (and interested to).

**Comment by donat** [comment]
Hi @Golvanig, Thanks for the offer! The timing is rather good as a blocking issue in the Kotlin tooling was resolved recently.

If you want to try the Kotlin build script support, the steps [above].(https://github.com/eclipse/buildship/issues/222#issuecomment-277469185) are still up-to-date.
Please note, that the language support we extend is provided by JetBrains and it is still in the pre-1.0 stage. 

If you have any feedback or need some help with the setup, feel free to ping us here.

**Comment by Golvanig** [comment]
Hello,
I just tried with a new clean neon.3 instance (J2EE package)
and installed the 2 plugins as described,
Imported my kotlin project (with build.gradle.kts)

OK: -- "kt" files are opened as usual 
KO: -- kts can't be opened**
I got this issue:
"Problems occurred when invoking code from plug-in: "org.eclipse.e4.ui.workbench.swt". (detailed under)
The run with build.gradle.kts is though operational (I just have very very basic features in) through buildship or gradle command line.
Kr,

-------------------------
java.lang.AbstractMethodError: org.eclipse.buildship.kotlin.GradleKotlinScriptTemplateProvider.getTemplateClasspath(Ljava/util/Map;)Ljava/lang/Iterable;
	at org.jetbrains.kotlin.core.model.ScriptTemplateProviderEPKt.makeScriptDefsFromEclipseTemplatesProviders(scriptTemplateProviderEP.kt:40)
	at org.jetbrains.kotlin.core.model.ScriptTemplateProviderEPKt.loadAndCreateDefinitionsByTemplateProviders(scriptTemplateProviderEP.kt:24)
	at org.jetbrains.kotlin.core.model.KotlinScriptEnvironment.<init>(KotlinEnvironment.kt:94)
	at org.jetbrains.kotlin.core.model.KotlinScriptEnvironment.<init>(KotlinEnvironment.kt:91)
	at org.jetbrains.kotlin.core.model.KotlinScriptEnvironment$Companion$environmentCreation$1.invoke(KotlinEnvironment.kt:125)
	at org.jetbrains.kotlin.core.model.KotlinScriptEnvironment$Companion$environmentCreation$1.invoke(KotlinEnvironment.kt:119)
	at org.jetbrains.kotlin.core.model.CachedEnvironment.getOrCreateEnvironment(CachedEnvironment.kt:40)
	at org.jetbrains.kotlin.core.model.KotlinScriptEnvironment$Companion.getEnvironment(KotlinEnvironment.kt:131)
	at org.jetbrains.kotlin.core.model.KotlinEnvironmentKt.getEnvironment(KotlinEnvironment.kt:67)
	at org.jetbrains.kotlin.ui.editors.highlighting.KotlinTokenScanner.createKtFile(KotlinTokenScanner.java:104)
	at org.jetbrains.kotlin.ui.editors.highlighting.KotlinTokenScanner.setRange(KotlinTokenScanner.java:62)
	at org.eclipse.jface.text.rules.DefaultDamagerRepairer.createPresentation(DefaultDamagerRepairer.java:173)
	at org.eclipse.jface.text.presentation.PresentationReconciler.createPresentation(PresentationReconciler.java:434)
	at org.eclipse.jface.text.presentation.PresentationReconciler.processDamage(PresentationReconciler.java:547)
	at org.eclipse.jface.text.presentation.PresentationReconciler.access$3(PresentationReconciler.java:545)
	at org.eclipse.jface.text.presentation.PresentationReconciler$InternalListener.inputDocumentChanged(PresentationReconciler.java:118)
	at org.eclipse.jface.text.TextViewer.fireInputDocumentChanged(TextViewer.java:2759)
	at org.eclipse.jface.text.TextViewer.setDocument(TextViewer.java:2800)
	at org.eclipse.jface.text.source.SourceViewer.setDocument(SourceViewer.java:634)
	at org.eclipse.jface.text.source.projection.ProjectionViewer.setDocument(ProjectionViewer.java:365)
	at org.eclipse.jface.text.source.SourceViewer.setDocument(SourceViewer.java:584)
	at org.eclipse.ui.texteditor.AbstractTextEditor.initializeSourceViewer(AbstractTextEditor.java:3995)
	at org.eclipse.ui.texteditor.AbstractTextEditor.createPartControl(AbstractTextEditor.java:3465)
	at org.eclipse.ui.texteditor.StatusTextEditor.createPartControl(StatusTextEditor.java:55)
	at org.eclipse.ui.texteditor.AbstractDecoratedTextEditor.createPartControl(AbstractDecoratedTextEditor.java:445)
	at org.eclipse.jdt.internal.ui.javaeditor.JavaEditor.createPartControl(JavaEditor.java:3094)
	at org.eclipse.jdt.internal.ui.javaeditor.CompilationUnitEditor.createPartControl(CompilationUnitEditor.java:1527)
	at org.jetbrains.kotlin.ui.editors.KotlinCommonEditor.createPartControl(KotlinCommonEditor.kt:96)
	at org.jetbrains.kotlin.ui.editors.KotlinScriptEditor.createPartControl(KotlinScriptEditor.kt:46)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityPart.createPartControl(CompatibilityPart.java:150)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityEditor.createPartControl(CompatibilityEditor.java:99)
	at org.eclipse.ui.internal.e4.compatibility.CompatibilityPart.create(CompatibilityPart.java:340)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.e4.core.internal.di.MethodRequestor.execute(MethodRequestor.java:55)
	at org.eclipse.e4.core.internal.di.InjectorImpl.processAnnotated(InjectorImpl.java:966)
	at org.eclipse.e4.core.internal.di.InjectorImpl.processAnnotated(InjectorImpl.java:931)
	at org.eclipse.e4.core.internal.di.InjectorImpl.inject(InjectorImpl.java:151)
	at org.eclipse.e4.core.internal.di.InjectorImpl.internalMake(InjectorImpl.java:375)
	at org.eclipse.e4.core.internal.di.InjectorImpl.make(InjectorImpl.java:294)
	at org.eclipse.e4.core.contexts.ContextInjectionFactory.make(ContextInjectionFactory.java:162)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.createFromBundle(ReflectionContributionFactory.java:105)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.doCreate(ReflectionContributionFactory.java:74)
	at org.eclipse.e4.ui.internal.workbench.ReflectionContributionFactory.create(ReflectionContributionFactory.java:56)
	at org.eclipse.e4.ui.workbench.renderers.swt.ContributedPartRenderer.createWidget(ContributedPartRenderer.java:129)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createWidget(PartRenderingEngine.java:975)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:651)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.safeCreateGui(PartRenderingEngine.java:757)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.access$0(PartRenderingEngine.java:728)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$2.run(PartRenderingEngine.java:722)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.createGui(PartRenderingEngine.java:706)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl$1.handleEvent(PartServiceImpl.java:104)
	at org.eclipse.e4.ui.services.internal.events.UIEventHandler$1.run(UIEventHandler.java:40)
	at org.eclipse.swt.widgets.Synchronizer.syncExec(Synchronizer.java:233)
	at org.eclipse.ui.internal.UISynchronizer.syncExec(UISynchronizer.java:145)
	at org.eclipse.swt.widgets.Display.syncExec(Display.java:5414)
	at org.eclipse.e4.ui.internal.workbench.swt.E4Application$1.syncExec(E4Application.java:211)
	at org.eclipse.e4.ui.services.internal.events.UIEventHandler.handleEvent(UIEventHandler.java:36)
	at org.eclipse.equinox.internal.event.EventHandlerWrapper.handleEvent(EventHandlerWrapper.java:201)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:197)
	at org.eclipse.equinox.internal.event.EventHandlerTracker.dispatchEvent(EventHandlerTracker.java:1)
	at org.eclipse.osgi.framework.eventmgr.EventManager.dispatchEvent(EventManager.java:230)
	at org.eclipse.osgi.framework.eventmgr.ListenerQueue.dispatchEventSynchronous(ListenerQueue.java:148)
	at org.eclipse.equinox.internal.event.EventAdminImpl.dispatchEvent(EventAdminImpl.java:135)
	at org.eclipse.equinox.internal.event.EventAdminImpl.sendEvent(EventAdminImpl.java:78)
	at org.eclipse.equinox.internal.event.EventComponent.sendEvent(EventComponent.java:39)
	at org.eclipse.e4.ui.services.internal.events.EventBroker.send(EventBroker.java:94)
	at org.eclipse.e4.ui.internal.workbench.UIEventPublisher.notifyChanged(UIEventPublisher.java:60)
	at org.eclipse.emf.common.notify.impl.BasicNotifierImpl.eNotify(BasicNotifierImpl.java:374)
	at org.eclipse.e4.ui.model.application.ui.impl.ElementContainerImpl.setSelectedElement(ElementContainerImpl.java:173)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.showElementInWindow(ModelServiceImpl.java:617)
	at org.eclipse.e4.ui.internal.workbench.ModelServiceImpl.bringToTop(ModelServiceImpl.java:581)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.delegateBringToTop(PartServiceImpl.java:770)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.bringToTop(PartServiceImpl.java:401)
	at org.eclipse.e4.ui.internal.workbench.PartServiceImpl.showPart(PartServiceImpl.java:1214)
	at org.eclipse.ui.internal.WorkbenchPage.busyOpenEditor(WorkbenchPage.java:3261)
	at org.eclipse.ui.internal.WorkbenchPage.access$25(WorkbenchPage.java:3176)
	at org.eclipse.ui.internal.WorkbenchPage$10.run(WorkbenchPage.java:3158)
	at org.eclipse.swt.custom.BusyIndicator.showWhile(BusyIndicator.java:70)
	at org.eclipse.ui.internal.WorkbenchPage.openEditor(WorkbenchPage.java:3153)
	at org.eclipse.ui.internal.WorkbenchPage.openEditor(WorkbenchPage.java:3117)
	at org.eclipse.ui.internal.WorkbenchPage.openEditor(WorkbenchPage.java:3107)
	at org.eclipse.ui.ide.IDE.openEditor(IDE.java:563)
	at org.eclipse.ui.ide.IDE.openEditor(IDE.java:518)
	at org.eclipse.jdt.internal.ui.javaeditor.EditorUtility.openInEditor(EditorUtility.java:360)
	at org.eclipse.jdt.internal.ui.javaeditor.EditorUtility.openInEditor_aroundBody0(EditorUtility.java:167)
	at org.eclipse.jdt.internal.ui.javaeditor.EditorUtility$AjcClosure1.run(EditorUtility.java:1)
	at org.jetbrains.kotlin.aspects.navigation.KotlinOpenEditorAspect.ajc$around$org_jetbrains_kotlin_aspects_navigation_KotlinOpenEditorAspect$1$d64491c4proceed(KotlinOpenEditorAspect.aj:16)
	at org.jetbrains.kotlin.aspects.navigation.KotlinOpenEditorAspect.ajc$around$org_jetbrains_kotlin_aspects_navigation_KotlinOpenEditorAspect$1$d64491c4(KotlinOpenEditorAspect.aj:29)
	at org.eclipse.jdt.internal.ui.javaeditor.EditorUtility.openInEditor(EditorUtility.java:164)
	at org.eclipse.jdt.ui.actions.OpenAction.run(OpenAction.java:256)
	at org.eclipse.jdt.ui.actions.OpenAction.run(OpenAction.java:221)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.dispatchRun(SelectionDispatchAction.java:271)
	at org.eclipse.jdt.ui.actions.SelectionDispatchAction.run(SelectionDispatchAction.java:249)
	at org.eclipse.jdt.internal.ui.navigator.OpenAndExpand.run(OpenAndExpand.java:50)
	at org.eclipse.ui.actions.RetargetAction.run(RetargetAction.java:225)
	at org.eclipse.ui.navigator.CommonNavigatorManager$2.open(CommonNavigatorManager.java:191)
	at org.eclipse.ui.OpenAndLinkWithEditorHelper$InternalListener.open(OpenAndLinkWithEditorHelper.java:46)
	at org.eclipse.jface.viewers.StructuredViewer$2.run(StructuredViewer.java:855)
	at org.eclipse.core.runtime.SafeRunner.run(SafeRunner.java:42)
	at org.eclipse.ui.internal.JFaceUtil$1.run(JFaceUtil.java:50)
	at org.eclipse.jface.util.SafeRunnable.run(SafeRunnable.java:173)
	at org.eclipse.jface.viewers.StructuredViewer.fireOpen(StructuredViewer.java:852)
	at org.eclipse.jface.viewers.StructuredViewer.handleOpen(StructuredViewer.java:1169)
	at org.eclipse.ui.navigator.CommonViewer.handleOpen(CommonViewer.java:451)
	at org.eclipse.jface.util.OpenStrategy.fireOpenEvent(OpenStrategy.java:278)
	at org.eclipse.jface.util.OpenStrategy.access$2(OpenStrategy.java:272)
	at org.eclipse.jface.util.OpenStrategy$1.handleEvent(OpenStrategy.java:313)
	at org.eclipse.swt.widgets.EventTable.sendEvent(EventTable.java:84)
	at org.eclipse.swt.widgets.Display.sendEvent(Display.java:5227)
	at org.eclipse.swt.widgets.Widget.sendEvent(Widget.java:1340)
	at org.eclipse.swt.widgets.Display.runDeferredEvents(Display.java:4561)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:4151)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine$4.run(PartRenderingEngine.java:1121)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.e4.ui.internal.workbench.swt.PartRenderingEngine.run(PartRenderingEngine.java:1022)
	at org.eclipse.e4.ui.internal.workbench.E4Workbench.createAndRunUI(E4Workbench.java:150)
	at org.eclipse.ui.internal.Workbench$5.run(Workbench.java:693)
	at org.eclipse.core.databinding.observable.Realm.runWithDefault(Realm.java:336)
	at org.eclipse.ui.internal.Workbench.createAndRunWorkbench(Workbench.java:610)
	at org.eclipse.ui.PlatformUI.createAndRunWorkbench(PlatformUI.java:148)
	at org.eclipse.ui.internal.ide.application.IDEApplication.start(IDEApplication.java:138)
	at org.eclipse.equinox.internal.app.EclipseAppHandle.run(EclipseAppHandle.java:196)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.runApplication(EclipseAppLauncher.java:134)
	at org.eclipse.core.runtime.internal.adaptor.EclipseAppLauncher.start(EclipseAppLauncher.java:104)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:388)
	at org.eclipse.core.runtime.adaptor.EclipseStarter.run(EclipseStarter.java:243)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.eclipse.equinox.launcher.Main.invokeFramework(Main.java:673)
	at org.eclipse.equinox.launcher.Main.basicRun(Main.java:610)
	at org.eclipse.equinox.launcher.Main.run(Main.java:1519)
	at org.eclipse.equinox.launcher.Main.main(Main.java:1492)

**Comment by Golvanig** [comment]
And
the icon of the kts files are still displayed with the ? question mark.
I get this error message, I think related to:

```
An internal error occurred during: "Compute launch button tooltip".
org.eclipse.buildship.kotlin.GradleKotlinScriptTemplateProvider.getTemplateClasspath(Ljava/util/Map;)Ljava/lang/Iterable;
```

**Comment by donat** [comment]
I think I found the problem. Buildship is using an experimental interface from Kotlin-Eclipse which hasn't been merged into the main branch. To make it work, you should install Kotlin-Eclipse from the following update site: https://teamcity.jetbrains.com/guestAuth/repository/download/Kotlin_EclipsePlugin/1079489:id/

I'll adjust the link in my post above.

**Comment by Golvanig** [comment]
Hello,
Thanks for the new update site.
I reloaded the new Kotlin_EclipsePlugin and I can open the kts files.
But ...
I do not see any improvement in the syntax highlighting, the error markers, ...
for instance in
"import groovy.lang.GroovyObject"
- import is bold
- and "groovy" is red underlined with Unresolved reference

**Comment by calvertdw** [comment]
I'm getting the same results on your branch as I am the latest release.

![image](https://user-images.githubusercontent.com/1161586/28496500-c21ff0ac-6f32-11e7-9e18-549835f97929.png)

Gradle 4.1-rc-1
Buildship 2.1.1
Kotlin-Eclipse 0.8.2

**Comment by calvertdw** [comment]
Is this project dead? Our users are flocking to IntelliJ. Should I wait on buildship to move all of our build scripts to Kotlin, or is this never gonna happen?

**Comment by oehme** [comment]
We're still alive, but there is not much that Buildship actually does - all the work happens in the Kotlin Eclipse plugin. @donat gave me a nice demo some time ago which worked properly, so I'm not sure if something got broken again or if there is some difference in your setup. @donat does it still work for you?

**Comment by donat** [comment]
It's been a while since we worked on the Kotlin integration. The main functionality is there, but there were a couple of rough edges. We reported the problems to JetBrains, but so far they didn't prioritize them. Besides, the project is largely undocumented so It's really hard to contribute fixes.

In any case, we could publish the Kotlin integration as an optional, incubating feature. This way users could give us and JetBrains more feedback. @oehme  WDYT?

**Comment by calvertdw** [comment]
Hey @donat! I'm back... XD. Would you mind publishing that incubating version? I'd like to play around with it and see where we're at. Dying to switch our Gradle scripts to Kotlin.

**Comment by donat** [comment]
@calvertdw You can install a snapshot version from [this update site](https://builds.gradle.org/repository/download/Tooling_Master_Eclipse_CrossVersionCoverage_CrossVersionBuildLinuxEclipse47Java9/11212628:id/update-site/) (credentials: guest/guest). Frankly, there's a lot of room for improvement in Kotlin-Eclipse. The tooling loads things synchronously when the editor is opened. Also, you have to open and close the editor when you apply a plugin to see proper completion. Nonetheless, if you find another problem you can report it here or at [the JetBrains issue tracker](https://youtrack.jetbrains.com/oauth?state=%2F).

**Comment by calvertdw** [comment]
Thanks @donat! I'll try to get some issues submitted to them.

**Comment by GFriedrich** [comment]
Hi guys,
did you receive an update from Jetbrains in the meantime?
Because now all Kotlin Eclipse tickets vanished from their YouTrack instance.

I've also put this question into their discourse instance. Let's see what they will answer: https://discuss.kotlinlang.org/t/youtrack-kotlin-eclipse-project-gone/8161

If you've any information, it would be nice to know them.
Thanks in advance.

**Comment by donat** [comment]
Not much has moved ever since. Last time I've checked the state of Kotlin-Eclipse it was still kind of unusable in production.

**Comment by fwgreen** [comment]
It seems the Kotlin team is looking into this:
https://youtrack.jetbrains.com/issue/KE-273

**Comment by donat** [comment]
@fwgreen Thanks for the update. I'm looking forward to have a stable Eclipse plugin for Kotlin.

**Comment by calvertdw** [comment]
@GFriedrich I understand why Jetbrains might not want to contribute this feature, but why not Gradle? Surely they want to decouple themselves from being an IntelliJ exclusive product. For instance, my team decided to use Gradle because it supports both IDEs.

**Comment by oehme** [comment]
> but why not Gradle? 

We're not in the business of building language support for IDEs ;) Plus, JetBrains is in a much better position to provide support for their own language. We hold the same position with Groovy btw.

> For instance, my team decided to use Gradle because it supports both IDEs.

We do, to the extent that the IDEs allow us. We are in close contact with JetBrains and will make sure that the Kotlin scripting situation improves.

**Comment by donat** [comment]
Not really. The existing Kotlin integration in Eclipse is simply too unstable to work with Gradle DSL. We'll coordinate with JetBrains how can we improve the situation. I'll close this issue for the time being and reopen when we have an update.

**Comment by chhex** [comment]
I totally appreciate the great work the gradle also kotlin developers are doing. But the situation is quite frustrating for the normal users: One of the main motivations to develop gradle kotlin, was to have better IDE support for gradle. Current situation: Gradle Kotlin not usable with Eclipse, Gradle Groovy is the way better option.

**Comment by sdavids** [comment]
Any update? -- And why is this issue closed?

@oehme Your stance is understandable, but 3 years later IntelliJ is still the only IDE with proper support for Kotlin DSL.

So if Gradle Kotlin DSL is the future™️ ...

**Comment by pri-me** [comment]
And another 3hree years gone, maybe my grandchildren will enjoy gradle kotlin dsl in eclipse :P

**Comment by sdavids** [comment]
https://blog.gradle.org/kotlin-dsl-is-now-the-default-for-new-gradle-builds

> April 13, 2023
> Kotlin DSL is now the default choice for new Gradle builds

**Comment by pri-me** [comment]
Rest assured that I do feel bad for complaining without offering solutions 😅
